src/lib/coda_lib/coda_lib.ml

```diff
@@ -475,8 +476,11 @@ let request_work t =
   if List.is_empty instances then None
   else Some {Snark_work_lib.Work.Spec.instances; fee}
 
+let next_proposal t = t.next_proposal
+
 let start t =
   Proposer.run ~logger:t.config.logger ~verifier:t.processes.verifier
+    ~set_next_proposal:(fun p -> t.next_proposal <- Some p)
```

Comment:
I feel like we should pass in the actual mutable ref of `next_proposal` rather than a callback. It would be easier to read in the long run.

Reply:
Callbacks are way more flexible than refs, general :+1: from me for preferring them over immediate mutation. Mutable fields are also slightly more efficient than having a ref, although I don't t think that's a relevant concern here.